File: packages/amplitude_experiments_flutter_ios/ios/Classes/AmplitudeExperimentsFlutterPlugin.swift
import Flutter
import UIKit

/// Flutter plugin for Amplitude Experiments on iOS.
///
/// This plugin registers the Pigeon-generated API handler to enable
/// communication between Flutter and the native Amplitude Experiments SDK.
public class AmplitudeExperimentsFlutterPlugin: NSObject, FlutterPlugin {
    private var apiImpl: AmplitudeExperimentsApiImpl?

    public static func register(with registrar: FlutterPluginRegistrar) {
        let instance = AmplitudeExperimentsFlutterPlugin()
        instance.apiImpl = AmplitudeExperimentsApiImpl()
        AmplitudeExperimentsApiSetup.setUp(
            binaryMessenger: registrar.messenger(),
            api: instance.apiImpl
        )
    }

    public func detachFromEngine(for registrar: FlutterPluginRegistrar) {
        AmplitudeExperimentsApiSetup.setUp(
            binaryMessenger: registrar.messenger(),
            api: nil
        )
        apiImpl = nil
    }
}

File: packages/amplitude_experiments_flutter_android/android/src/main/kotlin/dev/kaique/amplitude_experiments_flutter/AmplitudeExperimentsFlutterPlugin.kt
package dev.kaique.amplitude_experiments_flutter

import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * Flutter plugin for Amplitude Experiments on Android.
 *
 * This plugin registers the Pigeon-generated API handler to enable
 * communication between Flutter and the native Amplitude Experiments SDK.
 */
class AmplitudeExperimentsFlutterPlugin : FlutterPlugin {
    private var apiImpl: AmplitudeExperimentsApiImpl? = null

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        val context = flutterPluginBinding.applicationContext
        apiImpl = AmplitudeExperimentsApiImpl(context)
        AmplitudeExperimentsApi.setUp(flutterPluginBinding.binaryMessenger, apiImpl)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        AmplitudeExperimentsApi.setUp(binding.binaryMessenger, null)
        apiImpl = null
    }
}
